source/FreeRTOS_IP_Timers.c

```diff
@@ -55,16 +55,6 @@
 #include "NetworkBufferManagement.h"
 #include "FreeRTOS_Routing.h"
 #include "FreeRTOS_DNS.h"
-/*-----------------------------------------------------------*/
-
-/** @brief 'xAllNetworksUp' becomes pdTRUE as soon as all network interfaces have
- * been initialised. */
-/* MISRA Ref 8.9.1 [File scoped variables] */
-/* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-89 */
-/* coverity[misra_c_2012_rule_8_9_violation] */
-static BaseType_t xAllNetworksUp = pdFALSE;
```

Comment:
Instead of removing `xAllNetworksUp` maybe we could have a set/reset API to update the state of the variable when the network goes down maybe from `prvProcessNetworkDownEvent`.
This could avoid unnecesary looping through the interfaces whenever the network timer expires.

Reply:
I left `xAllNetworksUp` intact and added a Private API for setting it to pdFALSE.

Reply:
Yeah seems fine. You might have to add Unit Test for the new function you have added for code coverage.